TESTING.md

```diff
@@ -0,0 +1,291 @@
+```
+
+Related files should be located under `/.vib/app/goss`.
+
+### Test suite
+
+A GOSS test suite for a given application can be divided in two. One half contains the tests specifically manufactured to verify different aspects of the application and the other half is intended to verify that the container complies with Bitnami's best practices. As some of these tests are oftentimes almost identical between different apps, we have compiled some of them in a series of GOSS templates to unify and simplify their usage.
+
+#### GOSS Templates
+
+At their core, GOSS templates are just tests of similar nature put together on a `.yaml` file under the `.vib/common/goss/templates` folder.
+
+Every template is composed of:
+
+* A brief description of the tests' nature
+* Any needed vars it may use to run its tests
+* One or more tests
+
+To better understand them, let's see one of these templates:
+
+```yaml
+########################
+# Checks binaries are added to the $PATH
+# Needed vars:
+#   - .Vars.binaries (Array)
+########################
+command:
+  {{ range $binary := .Vars.binaries }}
+  check-{{ $binary }}-binary:
+    exec: which {{ $binary }}
+    exit-status: 0
+  {{ end }}
+```
+
+In the example above, we can see the template will execute a `which` command for every binary included in the `.Vars.binaries` array. These variables may be optional or required and must be defined in the `/.vib/app/goss/vars.yaml` file.
+
+There are many different templates, some of them focusing on verifying specific types of apps (those that use PHP, NGINX, etc.) or a particular best practice. In the same vein, new templates can be added if a particular group of tests is going to be used by several apps.
+
+#### Composition
+
+There are instances where it is not needed to create custom tests for a given app, where using the templates will suffice. There will also be suites that may require the use of testing files to properly verify the app. Generally, a test suite can be composed of the following files:
+
+```bash
+.vib/java/goss
+├── goss.yaml
+├── java.yaml
+├── testfiles
+│   └── HelloTest.jar
+└── vars.yaml
+```
+
+* [ ] The optional `app.yaml` file is where custom tests created just for the related app verification are included.
+* [ ] The `goss.yaml` file should include the list of used GOSS templates as well as the `app.yaml` file (if necessary).
+* [ ] The `vars.yaml` file should include every variable used in the templates.
+* [ ] The `testfiles` folder should include any external files used in the `app.yaml` tests.
+
+Not every suite will be composed of the same tests, as it will depend on the type of application, its Dockerfile, and the used compilation/configuration logic. The list below details each pillar that should be checked when creating a test suite as well as when to use some of the most common templates:
+
+* Dockerfile check:
+  * Does it include nami modules which contain binaries? If so, use the `check-binaries.yaml` GOSS template.
```

Comment:
I don't think this terminology is as clear as to us for external contributors. 

```suggestion
  * Does it include bitnami sub-modules which contain binaries? If so, use the `check-binaries.yaml` GOSS template.
```

Reply:
You are right, I'll change this to the one we actually use nowadays; _bitnami components_.